torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1389,13 +1403,96 @@ void ONNXShapeTypeInference(
 
 } // namespace
 
+std::pair<bool, bool> IsShapeReliable(Node* n) {
+  auto reliable = true;
+  auto complete = true;
+  for (auto input : n->inputs()) {
+    if (ConstantValueMap::HasTypeReliable(input->debugName())) {
+      auto reliable_value =
+          ConstantValueMap::GetTypeReliable(input->debugName()).value();
+      reliable = reliable && reliable_value;
+    } else {
+      reliable = false;
+    }
+    if (auto pt = input->type()->cast<TensorType>()) {
+      if (!pt->sizes().isComplete()) {
+        complete = false;
+      }
+    }
+  }
+  return std::make_pair(reliable, complete);
+}
+
+// There is no need to put onnx type here, but we need this
+// for some legacy tests when onnx_shape_inference=False.
+static std::unordered_set<std::string> nodeTypeReliableForTracer = {
+    "prim::ListConstruct",
+    "onnx::Cast",
+    "onnx::Constant",
+    "com.microsoft::Gelu"};
+
+void UpdateReliable(
+    torch::jit::Value* output,
+    const std::pair<bool, bool>& inferred_type_reliable) {
+  auto inferred = false;
+  if (ConstantValueMap::HasUseInferredType(output->debugName())) {
+    inferred =
+        ConstantValueMap::GetUseInferredType(output->debugName()).value();
+  }
+  auto isTypeReliableForTracer =
+      nodeTypeReliableForTracer.find(
+          output->node()->kind().toDisplayString()) !=
```

Comment:
nit: above comment for use c10::Symbol, and compare directly without cast to string.